File: apiauth_signature.php
<?php
//php apiauth_signature.php

//Today
echo(gmdate('D, d M Y H:i:s T'));

//Get Canonical String from Console
echo("Enter Canonical String (method-type,content-type,content-MD5,requestURI,timestamp)\n");
$canonical_string = readline();

//Get Secret Key from Console
echo("Enter Secret Key\n");
$secret_key = readline();

//Encoding 
$pass1 = hash_hmac('sha1', $canonical_string, $secret_key, true);
$signature = base64_encode($pass1);

//Return Signature
echo("Signature:\n");
echo($signature."\n");
?>

File: apiauth_signature.cs
//mcs apiauth_signature.cs
//mono apiauth_signature.exe
using System;
using System.Text;
using System.Security.Cryptography;
					
public class Program
{
	public static void Main()
	{
		//Get Canonical String from Console
		Console.WriteLine("Enter Canonical String (method-type,content-type,content-MD5,requestURI,timestamp)");
		string canonical_string = Console.ReadLine();

		//Get Secret Key from Console
		Console.WriteLine("Enter Secret Key");
		string secret_key = Console.ReadLine();

		//Encoding
		byte[] hashKey = Encoding.ASCII.GetBytes(secret_key);
		HMACSHA1 myhmacsha1 = new HMACSHA1(hashKey);
		byte[] hashCanonical = Encoding.ASCII.GetBytes(canonical_string);
		byte[] hashSignature = myhmacsha1.ComputeHash(hashCanonical);
		string signature = System.Convert.ToBase64String(hashSignature);

		//Return Signature
		Console.WriteLine("Signature:");
		Console.WriteLine(signature);
		
	}
}
